src/NUnitFramework/framework/Constraints/EqualNumericConstraint.cs

```diff
@@ -10,18 +8,19 @@ namespace NUnit.Framework.Constraints
     /// considered equal if both are null, or if both have the same
     /// value. NUnit has special semantics for some object types.
     /// </summary>
-#pragma warning disable CS3024 // Constraint type is not CLS-compliant
     public class EqualNumericConstraint<T> : EqualNumericWithoutUsingConstraint<T>, IEqualWithUsingConstraint<T>
-        where T : unmanaged, IConvertible, IEquatable<T>
```

Comment:
question: I understand why we added the overloads on `Is.EqualTo()`, but can you remind me why we needed to go back to `struct` here?

Related question about the need to make the constructor for this `internal`... Now that we've shipped a public constructor, is there a harm in letting people target it? I feel like the could also open some extensibility doors. 

For example, a `ComplexNumberEqualConstraint` which internally composes two separate `EqualNumericConstraint` instances.

Reply:
> why we needed to go back to `struct` here?

Unconstraint generics resulted in `Possible null reference` compile time error, see [build](https://github.com/nunit/nunit/actions/runs/12414128664)

I could have used `notnull`, but the only supported numeric types are value types and maybe the compiler can do something more knowing that.

> Related question about the need to make the constructor for this `internal`... Now that we've shipped a public constructor, is there a harm in letting people target it? I feel like the could also open some extensibility doors.

Initially I made it `internal` to prevent it being called with unsupported types.
As I have now added a `Guard` protecting it that is less of an issue, so if you prefer to have it `public`, that is fine by me.